android/source/VideoLocker/src/org/edx/mobile/view/custom/URLInterceptorWebViewClient.java

```diff
@@ -0,0 +1,175 @@
+     * @param actionListener
+     */
+    public void setActionListener(IActionListener actionListener) {
+        this.actionListener = actionListener;
+    }
+
+    /**
+     * Sets up the WeView, applies minimal required settings and
+     * sets this class itself as WebViewClient.
+     * @param webView
+     */
+    private void setupWebView(WebView webView) {
+        WebSettings settings = webView.getSettings();
+        settings.setJavaScriptEnabled(true);
+        settings.setLoadWithOverviewMode(true);
+        settings.setBuiltInZoomControls(false);
+        settings.setSupportZoom(true);
+        webView.setWebViewClient(this);
+    }
+
+    @Override
+    public boolean shouldOverrideUrlLoading(WebView view, String url) {
+        try {
+            if (actionListener == null) {
+                logger.warn("you have not set IActionLister to this WebViewClient, " +
+                        "you might miss some event");
+            }
+
+            if (isCourseInfoLink(url)) {
+                // we handled this URL
+                return true;
+            } else if(isEnrollLink(url)) {
+                // we handled this URL
+                return true;
+            } else if (isExternalLink(url)) {
+                // open URL in external web browser
+                // return true means the host application handles the url
+                // this should open the URL in the browser
+                return true;
+            } else {
+                // return false means the current WebView handles the url.
+                return false;
+            }
+        } catch(Exception ex) {
+            logger.error(ex);
+        }
+        return super.shouldOverrideUrlLoading(view, url);
+    }
+
+    /**
+     * Checks if the URL pattern matches with that of COURSE_INFO URL.
+     * Extracts path_id from the URL and gives a callback to the registered
+     * action listener with path_id parameter.
+     * Returns true if pattern matches with COURSE_INFO URL pattern and callback succeeds with
+     * extracted parameter, false otherwise.
+     * @param strUrl
+     * @return
+     */
+    boolean isCourseInfoLink(String strUrl) {
+        try {
+            if (strUrl.startsWith(URL_TYPE_COURSE_INFO)) {
+                Uri uri = Uri.parse(strUrl);
+                String pathId = uri.getQueryParameter(PARAM_PATH_ID);
+
+                if (actionListener != null) {
+                    actionListener.onClickCourseInfo(pathId);
+                    return true;
+                }
+            }
+        } catch(Exception ex) {
+            logger.error(ex);
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns true if the pattern of the url matches with that of EXTERNAL URL pattern,
+     * false otherwise.
+     * @param url
+     * @return
+     */
+    boolean isExternalLink(String url) {
+        // TODO: update this method when form of external URL is determined
```

Comment:
An external link is anything that's not to "webview.edx.org". But, instead of hardcoding that, or at least adding it as a config, I think we can just compare its host to the host of the currently loaded URL?